### PR TITLE
feat(compat): expose compatibility metrics and CLI worker version stats (#1527)

### PR DESCRIPTION
**Pull request by lzjqsdd**
# Summary

Adds the observability companion for the compatibility checker and version
reporting introduced in the 0.4.x series: the master now exposes per-peer
compatibility verdicts and enforce-mode rejections as Prometheus metrics, and
the CLI can summarize the release-version distribution across live workers.

# Issue Describe / Design

Related to #1527 (component version management) — sub-task T12 deliverables:
compatibility matrix, CLI worker version statistics, observability metrics.

Design decisions:

- Metrics are recorded at the single point where compatibility verdicts are
  already evaluated (`MasterHandler::check_peer_compatibility`), so every
  worker heartbeat and client handshake is observable without touching each
  call site.
- Verdicts are exposed as `GaugeVec`s per peer (label set = peer id + verdict),
  so a mixed-version cluster shows exactly which peers are legacy or
  incompatible at a glance; `compat_enforce_rejected_total` counts rejections
  for alerting.
- The CLI version statistic prefers the structured `component_info` reported on
  heartbeat (T7+) and falls back to the legacy `software_version` string, so it
  works against both new and old workers.

# Changes

| Module / File | Change | Impact on existing behavior |
| ------------- | ------ | --------------------------- |
| `curvine-master/src/master/master_metrics.rs` | Add `compat_worker_verdict`, `compat_client_verdict` gauge vectors and `compat_enforce_rejected_total` counter vector | None (new metrics only) |
| `curvine-master/src/master/master_handler.rs` | Record verdicts and rejections in `check_peer_compatibility`; add `verdict_label` helper | None (diagnose/enforce behavior unchanged) |
| `curvine-cli/src/cmds/node.rs` | Add `cv node --versions` with `count_worker_versions` + unit tests | None (new command only) |

# Test verified

| Test case | Result | Notes |
| --------- | ------ | ----- |
| `make format` | PASS | clippy + fmt |
| `cargo test -p curvine-master --lib master_handler` | PASS | 8 tests incl. `compatibility_metrics_record_verdicts_and_rejections` |
| `cargo test -p curvine-cli` | PASS | 28 tests incl. 3 new `count_worker_versions` tests |
| `cargo check -p curvine-master -p curvine-cli` | PASS | |

# Dependencies

- Related PRs: None
- Related issues: #1527
- Blocks / blocked by: None

### CODE REVIEW

File: curvine-cli/src/cmds/node.rs [hunk 1]
```diff
@@ -274,7 +279,90 @@ impl NodeCommand {
...
+                    .and_then(|v| v.as_str())
+                {
+                    if !version.is_empty() {
+                        *version_counts.entry(version.to_string()).or_insert(0) += 1;
+                        continue;
+                    }
+                }
+                // component_info exists but empty -> treat as legacy
+                legacy_count += 1;
+            } else {
```
[the remark this answers is not in the document]
Comment (lzjqsdd) [reply curvine-cli/src/cmds/node.rs]: Thanks. Fixed in a12f1615: when `component_info` is present but `release_version` is missing or empty, `count_worker_versions` now falls back to the legacy `software_version` display string instead of counting the worker as legacy, matching the docstring. Replaced the old test with `count_worker_versions_falls_back_when_component_info_lacks_release_version` (covers both missing and empty release_version) and added `count_worker_versions_counts_legacy_when_no_version_data_at_all`.

File: curvine-master/src/master/master_handler.rs [hunk 2]
```diff
@@ -644,7 +646,24 @@ impl MasterHandler {
...
+            metrics
+                .compat_worker_verdict
+                .with_label_values(&[worker_id, verdict_label])
+                .set(1);
+        } else {
+            metrics
+                .compat_client_verdict
+                .with_label_values(&[dedup_key, verdict_label])
+                .set(1);
+        }
```
[the remark this answers is not in the document]
Comment (lzjqsdd) [reply curvine-master/src/master/master_handler.rs]: Thanks. Fixed in a1fa0c87: `check_peer_compatibility` now iterates all verdict labels for the peer and sets the current one to 1 while clearing every other label, so only one label per peer is active at a time and stale series no longer stick at 1. Added a regression assertion in `compatibility_metrics_record_verdicts_and_rejections` covering a peer whose verdict changes.

**Review by szbr00 — APPROVED** [review]
